File: solutions/852. Peak Index in a Mountain Array/0852_1.cpp
// Time complexity: O(⁡n)
// Space complexity: O(1)
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int i = 0;
        while (arr[i] < arr[i + 1]) {
            i++;
        }
        return i;
    }
};

File: solutions/0852. Peak Index in a Mountain Array/0852_1.py
# Time complexity: O(⁡n)
# Space complexity: O(1)
class Solution: 
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        i = 0
        while arr[i] < arr[i + 1]:
            i += 1

        return i